components/core/src/Utils.cpp

```diff
@@ -213,3 +219,124 @@ ErrorCode read_list_of_paths (const string& list_path, vector<string>& paths) {
+            schema_ast->m_delimiters.get());
+    if (!lexer.m_symbol_id.empty()) {
+        throw std::runtime_error("Error: symbol_ids initialized before setting enum symbol_ids");
+    }
+    /// TODO: this is a copy of other code
+    lexer.m_symbol_id[log_surgeon::cTokenEnd] = (int) log_surgeon::SymbolID::TokenEndID;
+    lexer.m_symbol_id[log_surgeon::cTokenUncaughtString] =
+            (int) log_surgeon::SymbolID::TokenUncaughtStringID;
+    lexer.m_symbol_id[log_surgeon::cTokenInt] = (int) log_surgeon::SymbolID::TokenIntId;
+    lexer.m_symbol_id[log_surgeon::cTokenFloat] = (int) log_surgeon::SymbolID::TokenFloatId;
+    lexer.m_symbol_id[log_surgeon::cTokenFirstTimestamp] = (int) log_surgeon::SymbolID::TokenFirstTimestampId;
+    lexer.m_symbol_id[log_surgeon::cTokenNewlineTimestamp] = (int) log_surgeon::SymbolID::TokenNewlineTimestampId;
+    lexer.m_symbol_id[log_surgeon::cTokenNewline] = (int) log_surgeon::SymbolID::TokenNewlineId;
+
+    lexer.m_id_symbol[(int) log_surgeon::SymbolID::TokenEndID] = log_surgeon::cTokenEnd;
+    lexer.m_id_symbol[(int) log_surgeon::SymbolID::TokenUncaughtStringID] =
+            log_surgeon::cTokenUncaughtString;
+    lexer.m_id_symbol[(int) log_surgeon::SymbolID::TokenIntId] = log_surgeon::cTokenInt;
+    lexer.m_id_symbol[(int) log_surgeon::SymbolID::TokenFloatId] = log_surgeon::cTokenFloat;
+    lexer.m_id_symbol[(int) log_surgeon::SymbolID::TokenFirstTimestampId] =
+            log_surgeon::cTokenFirstTimestamp;
+    lexer.m_id_symbol[(int) log_surgeon::SymbolID::TokenNewlineTimestampId] =
+            log_surgeon::cTokenNewlineTimestamp;
+    lexer.m_id_symbol[(int) log_surgeon::SymbolID::TokenNewlineId] = log_surgeon::cTokenNewline;
+
+    /// TODO: figure out why this needs to be specially added
+    lexer.add_rule(lexer.m_symbol_id["newLine"],
+                   std::move(std::make_unique<log_surgeon::finite_automata::RegexASTLiteral<
+                                              log_surgeon::finite_automata::RegexNFAByteState>>(
+            log_surgeon::finite_automata::RegexASTLiteral<
+                    log_surgeon::finite_automata::RegexNFAByteState>('\n'))));
+
+    if (delimiters_ptr != nullptr) {
+        lexer.add_delimiters(delimiters_ptr->m_delimiters);
+    }
+    for (std::unique_ptr<log_surgeon::ParserAST> const& parser_ast: schema_ast->m_schema_vars) {
+        auto* rule = dynamic_cast<log_surgeon::SchemaVarAST*>(parser_ast.get());
+
+        if ("timestamp" == rule->m_name) {
+            continue;
+        }
+
+        if (lexer.m_symbol_id.find(rule->m_name) == lexer.m_symbol_id.end()) {
+            lexer.m_symbol_id[rule->m_name] = lexer.m_symbol_id.size();
+            lexer.m_id_symbol[lexer.m_symbol_id[rule->m_name]] = rule->m_name;
+        }
+
+        // transform '.' from any-character into any non-delimiter character
+        rule->m_regex_ptr->remove_delimiters_from_wildcard(delimiters_ptr->m_delimiters);
+
+        /// TODO: this error function is a copy
```

Comment:
same as above, where and why it is a copy

Reply:
Duplicated from log_surgeon, added todo to create SearchParser in log_surgeon to remove duplication